exonum-java-binding-core/src/main/java/com/exonum/binding/storage/indices/ProofMapIndexProxy.java

```diff
@@ -203,33 +204,50 @@ public V get(K key) {
   private native byte[] nativeGet(long nativeHandle, byte[] key);
 
   /**
-   * Returns a proof that there is a value mapped to the specified key or
-   * that there is no such mapping.
+   * Returns a proof that there are values mapped to the specified keys or that there are no such
+   * mappings.
    *
    * @param key a proof map key which might be mapped to some value, must be 32-byte long
-   * @throws IllegalStateException  if this map is not valid
-   * @throws IllegalArgumentException if the size of the key is not 32 bytes
+   * @param otherKeys other proof map keys which might be mapped to some values, each must be
+   * 32-byte long
+   * @throws IllegalStateException if this map is not valid
+   * @throws IllegalArgumentException if the size of any of the keys is not 32 bytes
    */
-  public UncheckedMapProof getProof(K key) {
-    byte[] dbKey = keySerializer.toBytes(key);
-    return nativeGetProof(getNativeHandle(), dbKey);
+  public UncheckedMapProof getProof(K key, K... otherKeys) {
+    if (otherKeys.length == 0) {
+      byte[] dbKey = keySerializer.toBytes(key);
+      return nativeGetProof(getNativeHandle(), dbKey);
+    } else {
+      List<K> keyList = Arrays.asList(otherKeys);
+      keyList.add(key);
```

Comment:
I am not sure it will work (asList returns a fixed size array):

https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#asList(T...)

Do we test this?

Reply:
Thanks! I've added tests to cover all ways to get a proof.

Reply:
also, you can simply do `Arrays.asList(key, otherKeys)`

Reply:
It won't work like that because of incompatible types. The solution to that is to add a helper `appendKey` function:
```
List<? extends K> keys = Arrays.asList(appendKey(otherKeys, key));
```
```
  private static <K> K[] appendKey(K[] otherKeys, K key) {
    final int otherKeysSize = otherKeys.length;
    otherKeys = java.util.Arrays.copyOf(otherKeys, otherKeysSize + 1);
    otherKeys[otherKeysSize] = key;
    return otherKeys;
  }
```
But is it better than the existing solution?

Reply:
The present is better IMO